docs/reference/modules/node.asciidoc

```diff
@@ -1,32 +1,48 @@
-balancers that take part in some of different API processing. Lets take
-an example:
+*data nodes* store indexed data locally and make it available for
+further manipulation. Storing data locally basically means that
+shards of different indices are allowed to be allocated on that
+node. The default is `node.data: true`, but can also be set to
+`node.data: false` in the configuration file.
 
-We can start a whole cluster of data nodes which do not even start an
-HTTP transport by setting `http.enabled` to `false`. Such nodes will
-communicate with one another using the
-<<modules-transport,transport>> module. In front
-of the cluster we can start one or more "non data" nodes which will
-start with HTTP enabled. All HTTP communication will be performed
-through these "non data" nodes.
+*master nodes* control cluster state, including changes such as node
+addition or removal, index creation and deletion and mapping updates.
+While there may be multiple nodes with this setting enabled, in
+any cluster there can be only one _active master node_ at a time and
+if the current master becomes unavailable then the other master nodes
+can re-elect a new master, ensuring cluster continuity. If you have
+dedicated master nodes then you should not send index or query
+requests to them, this ensures they can maintain the cluster state
+with little to no interruption, ensuring the cluster remains as
+stable as possible. The default is `node.master: true` and
+can also be set to `node.master: false` in the configuration file.
 
-The benefit of using that is first the ability to create smart load
-balancers. These "non data" nodes are still part of the cluster, and
-they redirect operations exactly to the node that holds the relevant
-data. The other benefit is the fact that for scatter / gather based
-operations (such as search), these nodes will take part of the
-processing since they will start the scatter process, and perform the
-actual gather processing.
+*client nodes* create smart load balancers that take part in some of
+the different index and querying processes.  These "non data" nodes
+are still part of the cluster, and they redirect operations exactly
+to the node that holds the relevant data. The other benefit is the
+fact that for scatter / gather based operations (such as search),
+these nodes will take part of the processing since they will start
+the scatter process, and perform the actual gather processing. The
+default is `node.client: true` and can also be set to `node.master: false`
+in the configuration file. A node with `node.client: true` also implies
+that `node.master: false` and `node.data: false` is set, even if not
+specified in the configuration file.
 
-This relieves the data nodes to do the heavy duty of indexing and
-searching, without needing to process HTTP requests (parsing), overload
-the network, or perform the gather processing.
+Lets take an example:
+We have a cluster with 15 nodes. We decide to have 3 dedicated
```

Comment:
I wonder about this example. 3 dedicated master node is very typical. 2 dedicated clients isn't. Maybe its worth mentioning that?